app/models/manager_refresh/inventory_collection.rb

```diff
@@ -146,6 +149,11 @@ def object_index_with_keys(keys, object)
       keys.map { |attribute| object.public_send(attribute).to_s }.join(stringify_joiner)
     end
 
+    def find_or_build(manager_uuid)
+      # FIXME: splat manager_ref
+      data_index[manager_uuid] || build({ manager_ref.first => manager_uuid })
```

Comment:
then the same would be for lazy_find_by

we have few places now, where I could use separate index, but then it can return multiple results, there is only one unique index, which is manager_ref